keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/FloorVertex.java

```diff
@@ -16,16 +12,6 @@
      * @param inputVertex the vertex to be floor'd
      */
     public FloorVertex(DoubleVertex inputVertex) {
-        super(inputVertex.getShape(), inputVertex);
-    }
-
-    @Override
-    protected DoubleTensor op(DoubleTensor a) {
-        return a.floor();
-    }
-
-    @Override
-    public DualNumber calculateDualNumber(Map<Vertex, DualNumber> dualNumbers) {
-        throw new UnsupportedOperationException();
```

Comment:
So what happens now that we're not throwing an exception on Vertices that can't have their dual number calculated? Does it silently fail or will it still break?

Reply:
It means we're enforcing it at compile time instead of runtime (i.e. this class does not implement `Differentiable`).

If you were to try to cast an object of this type to a `Differentiable` you'd get a `ClassCastException`.